plugins/catalog/src/components/ComponentPage/ComponentPage.tsx

```diff
@@ -53,12 +54,13 @@ const ComponentPage: FC<ComponentPageProps> = ({ match, history }) => {
   const [removingPending, setRemovingPending] = useState(false);
   const showRemovalDialog = () => setConfirmationDialogOpen(true);
   const hideRemovalDialog = () => setConfirmationDialogOpen(false);
-  const componentName = match.params.name;
+  const { optionalNamespaceAndName, kind } = match.params;
+  const [name, namespace] = optionalNamespaceAndName.split(':').reverse();
```

Comment:
I think it could be good to fall back to the default `'default'` early, as in `namespace = 'default'` here, and then require the namespace to be set throughout the rest of logic. That way we don't end up with a bunch of `namespace ?? 'default'` all over the place.

Reply:
Currently this defaulting is happening only in the catalog API, which seems like a nice single place to handle that defaulting. And other places should just pass whatever they have for a namespace forward. Same goes for the backend - the only place we are getting the namespace is though the catalog API, where the namespace will be always present.

Reply:
Yeah true, especially if the catalog API ends up being the only boundary for it on the frontend. It would actually be pretty bad to have a bunch of `if (namespace === 'default')` for places where we don't want to show the namespace if it's default.